core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -608,7 +612,13 @@ private[spark] class SparkSubmit extends Logging {
           (deployMode & opt.deployMode) != 0 &&
           (clusterManager & opt.clusterManager) != 0) {
         if (opt.clOption != null) { childArgs += (opt.clOption, opt.value) }
```

Comment:
Just curious, I'm not sure that's a valid case, but would it be better to apply the same for `childArgs`?